File: auth-management/auth-model/src/main/java/com/pulse/air/auth/contract/AuthService.java
package com.pulse.air.auth.contract;

import java.io.IOException;

import com.pulse.air.auth.model.auth.LoginRequest;
import com.pulse.air.auth.model.user.UserRequest;
import com.pulse.air.common.model.ApiException;
import com.pulse.air.common.model.ApiRequest;
import com.pulse.air.common.model.ApiResponse;

public interface AuthService {

	public ApiResponse<String> login(final ApiRequest<LoginRequest> request) throws ApiException;

	public ApiResponse<String> register(final ApiRequest<UserRequest> request) throws ApiException;

	public ApiResponse<Boolean> validateToken(final ApiRequest<String> request) throws IOException;
}

File: airpulse-ui/src/app/pages/auth/services/auth.service.ts
import { HttpClient } from "@angular/common/http";
import { devEnvironment } from "src/environments/devenv";
import { ApiResponse } from "../../shared/model/api-response";
import { Observable, catchError, map, of } from "rxjs";
import { LoginRequest } from "../model/login";
import { UserRequest } from "../model/user";
import { Injectable } from "@angular/core";
import { LoaderService } from "../../shared/services/loader.service";

@Injectable({
  providedIn: 'root',
})
export class AuthService {
  protected constructor(
    protected http: HttpClient, protected loader: LoaderService
  ) { }
  url = `${devEnvironment.baseUrl}/auth`;

  validateToken(token: string): Observable<boolean> {
    return this.http.get<ApiResponse<boolean>>(`${this.url}/auth/validate-token`).pipe(
      map((response) => {
        return response['data'];
      })
    );
  }

  login(entity: LoginRequest): Observable<string> {
    this.loader.show();
    return this.http.post<ApiResponse<string>>(`${this.url}/auth/login`, entity).pipe(
      map((response) => {
        this.loader.hide();
        return response['data'];
      })
    );
  }

  register(entity: UserRequest): Observable<string> {
    //this.loader.show();
    return this.http.post<ApiResponse<string>>(`${this.url}/auth/register`, entity).pipe(
      map((response) => {
        //this.loader.hide();
        return response['data'];
      })
    );
  }

}
